tests/testapp/tests/test_api.py

```diff
@@ -584,3 +586,206 @@ def test_inactive_transfersession_cannot_be_deleted(self):
+        self.default_pull_filter = str(self.sub_subset_cert1_with_key.get_scope().read_filter)
+
+    def build_buffer_item(self, **kwargs):
+
+        if "transfer_session" not in kwargs:
+            assert "filter" in kwargs and "push" in kwargs
+            t_sess_req = self.make_transfersession_creation_request(**kwargs)
+            t_sess_id = json.loads(t_sess_req.content.decode())["id"]
+            kwargs["transfer_session"] = TransferSession.objects.get(id=t_sess_id)
+
+        client_cert = kwargs["transfer_session"].sync_session.remote_certificate
+        server_cert = kwargs["transfer_session"].sync_session.local_certificate
+        push = kwargs["transfer_session"].incoming
+
+        filt = client_cert.get_scope().write_filter if push else client_cert.get_scope().read_filter
+        partition = filt._filter_tuple[0] + ":furthersubpart"
+
+        data = {
+            "profile": kwargs.get("profile", kwargs["transfer_session"].sync_session.profile),
+            "serialized": kwargs.get("serialized", '{"test": 99}'),
+            "deleted": kwargs.get("deleted", False),
+            "last_saved_instance": kwargs.get("last_saved_instance", uuid.uuid4().hex),
+            "last_saved_counter": kwargs.get("last_saved_counter", 179),
+            "partition": kwargs.get("partition", partition),
+            "source_id": kwargs.get("source_id", uuid.uuid4().hex),
+            "model_name": kwargs.get("model_name", "contentsummarylog"),
+            "conflicting_serialized_data": kwargs.get("conflicting_serialized_data", ""),
+            "model_uuid": kwargs.get("model_uuid", None),
+            "transfer_session": kwargs["transfer_session"],
+        }
+
+        if not data["model_uuid"]:
+            Model = _profile_models[data["profile"]][data["model_name"]]
+            data["model_uuid"] = Model.compute_namespaced_id(data["partition"], data["source_id"], data["model_name"])
+
+        return Buffer(**data)
+
+    def make_buffer_post_request(self, buffers, expected_status=201):
+        serialized_recs = BufferSerializer(buffers, many=True)
+        response = self.client.post(reverse('buffers-list'), serialized_recs.data, format='json')
+        self.assertEqual(response.status_code, expected_status)
+        if expected_status == 201:
+            # check that the buffer items were created
+            self.assertEqual(Buffer.objects.count(), len(buffers))
+        else:
+            # check that the buffer items were not created
+            self.assertEqual(Buffer.objects.count(), 0)
+
+    def test_push_valid_buffer_chunk(self):
+        rec_1 = self.build_buffer_item(push=True, filter=self.default_push_filter)
+        rec_2 = self.build_buffer_item(transfer_session=rec_1.transfer_session)
+        rec_3 = self.build_buffer_item(transfer_session=rec_1.transfer_session)
+        self.make_buffer_post_request([rec_1, rec_2, rec_3], expected_status=201)
+
+    def test_push_with_invalid_model_uuid(self):
```

Comment:
Do it right or don't do it at all! :1st_place_medal: 

Why go on? What's left to live for, if a record has failed ye?